core/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -282,7 +282,10 @@ public FailOnRewriteQueryBuilder() {
 
         @Override
         protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) {
-            throw new IllegalStateException("Fail on rewrite phase");
+            if (queryShardContext.convertToShardContext() != null) {
```

Comment:
I pushed https://github.com/elastic/elasticsearch/commit/682abb90eed6ed3752bbc931f5a9e2aa6416e2bc to master for this